code/parse/parselo.cpp

```diff
@@ -1665,30 +1666,94 @@ int maybe_convert_foreign_character(int ch)
 }
 
 // Goober5000
-void maybe_convert_foreign_characters(char *line)
+// Yarn - The capacity of out must be at least the value returned by
+// get_converted_string_length(in) (plus one if add_null is true).
+// Returns the number of characters written to out.
+size_t maybe_convert_foreign_characters(char *in, char *out, bool add_null)
```

Comment:
1. Since `in` isn't changed anymore it should be declared `const`.
2. This should either have a separate parameter that specified the length of the buffer or just use an `SCP_string` to avoid buffer overrun errors.